src/replication.c

```diff
@@ -967,6 +900,77 @@ update_replica_entry(spec_t *spec, replica_t *replica, int iofd, char *replicaip
 	return replica;
 }
 
+static void
```

Comment:
did you picked changes from other pending PR from @payes ?

Reply:
Yes. To avoid conflicts, I pulled Payes changes for cleanup_replica so we can have a single version of cleanup_replica.